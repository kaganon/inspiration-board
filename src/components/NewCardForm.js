import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';
import './NewCardForm.css';

const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog"]


class NewCardForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      text: '',
      emoji: ''
    };
  }



  onInputChange = (event) => {
    console.log("In input change");

    const field = event.target.name;
    const value = event.target.value;

    const newState = {};
    newState[field] = value;

    this.setState(newState);
  };

  onFormSubmit = (event) => {
    event.preventDefault();

    const hasEmoji = this.state.emoji ? this.state.emoji : '';

    const newCard = {
      text: this.state.text,
      emoji: hasEmoji,
    };

    this.setState({
      text: '',
      emoji: ''
    });

    this.props.addCardCallback(newCard);
  };


  render() {
    return (
      <form
        className="new-card-form"
        >
        <h3 className="new-card-form__header">Add a Card </h3>
        <div>
          <form
            className="new-card-form"
            onSubmit={this.onFormSubmit}
            >
          <label htmlFor="text" className="new-card-form__form-label">
            Text
          </label>
          <textarea
            name="text"
            value={this.state.text}
            className="new-card-form__form"
            />
        </form>
        </div>
      </form>

    )
  }



}

export default NewCardForm;
