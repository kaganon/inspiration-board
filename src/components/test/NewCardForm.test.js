import React from 'react';
import NewCardForm from '../NewCardForm';
import { shallow } from 'enzyme';


describe('NewCardForm', () => {
  test('that it matches an existing snapshot', () => {
    // Arrange
    const wrapper = shallow(
      <NewCardForm addCardCallback={() => {}} />
    ); 

    // Assert
    expect(wrapper).toMatchSnapshot();
  });
});
