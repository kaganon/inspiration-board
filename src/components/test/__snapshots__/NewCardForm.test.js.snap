// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCardForm that it matches an existing snapshot 1`] = `
<div
  className="new-card-form"
>
  <h3
    className="new-card-form__header"
  >
    Give me some inspo!
  </h3>
  <div>
    <form
      className="new-card-form__form"
      onSubmit={[Function]}
    >
      <div>
        <label
          className="new-card-form__form-label"
          htmlFor="text"
        >
          Text
        </label>
      </div>
      <textarea
        className="new-card-form__form-textarea"
        name="text"
        onChange={[Function]}
        value=""
      />
      <label
        htmlFor="emoji-select"
      >
        Emoji
      </label>
      <select
        className="new-card-form__form-select"
        id="emoji-select"
        name="emoji"
        onChange={[Function]}
      >
        <option
          key="0"
        />
        <option
          key="1"
        >
          😍
        </option>
        <option
          key="2"
        >
          🍺
        </option>
        <option
          key="3"
        >
          👏
        </option>
        <option
          key="4"
        >
          💖
        </option>
        <option
          key="5"
        >
          😻
        </option>
        <option
          key="6"
        >
          🐶
        </option>
        <option
          key="7"
        >
          💩
        </option>
        <option
          key="8"
        >
          🍪
        </option>
        <option
          key="9"
        >
          🍩
        </option>
      </select>
      <input
        className="new-card-form__form-button"
        name="submit"
        type="submit"
        value="Add Card"
      />
    </form>
  </div>
</div>
`;
